File: lib/data/models/hospital_model.dart
/// Hospital model
class Hospital {
  final String id;
  final String name;
  final String imageUrl;
  final double rating;
  final String address;
  final String? phone;
  final bool emergency;
  final List<String> specialties;
  final int beds;
  final String type;

  Hospital({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.rating,
    required this.address,
    this.phone,
    this.emergency = false,
    this.specialties = const [],
    this.beds = 0,
    required this.type,
  });

  factory Hospital.fromJson(Map<String, dynamic> json) {
    return Hospital(
      id: json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      imageUrl: json['image'] ?? json['imageUrl'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      address: json['address'] ?? '',
      phone: json['phone'],
      emergency: json['emergency'] ?? false,
      specialties: (json['specialties'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
      beds: json['beds'] ?? 0,
      type: json['type'] ?? 'General',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'image': imageUrl,
      'rating': rating,
      'address': address,
      'phone': phone,
      'emergency': emergency,
      'specialties': specialties,
      'beds': beds,
      'type': type,
    };
  }
}

File: backend-new/src/modules/hospitals/hospital.model.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export interface IHospital extends Document {
  _id: Types.ObjectId;
  name: string;
  email: string;
  phone: string;
  address: {
    street: string;
    city: string;
    state: string;
    zipCode: string;
    coordinates?: {
      latitude: number;
      longitude: number;
    };
  };
  description?: string;
  imageUrl?: string;
  specialties: string[];
  facilities: string[];
  departments: string[];
  emergencyServices: boolean;
  rating: number;
  reviewCount: number;
  openingHours: {
    start: string;
    end: string;
  };
  beds: {
    total: number;
    available: number;
  };
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const HospitalSchema: Schema = new Schema(
  {
    name: {
      type: String,
      required: [true, 'Hospital name is required'],
      trim: true,
    },
    email: {
      type: String,
      required: [true, 'Email is required'],
      unique: true,
      lowercase: true,
    },
    phone: {
      type: String,
      required: [true, 'Phone number is required'],
    },
    address: {
      street: {
        type: String,
        required: true,
      },
      city: {
        type: String,
        required: true,
      },
      state: {
        type: String,
        required: true,
      },
      zipCode: {
        type: String,
        required: true,
      },
      coordinates: {
        latitude: Number,
        longitude: Number,
      },
    },
    description: {
      type: String,
    },
    imageUrl: {
      type: String,
    },
    specialties: [{
      type: String,
    }],
    facilities: [{
      type: String,
    }],
    departments: [{
      type: String,
    }],
    emergencyServices: {
      type: Boolean,
      default: true,
    },
    rating: {
      type: Number,
      default: 0,
      min: 0,
      max: 5,
    },
    reviewCount: {
      type: Number,
      default: 0,
    },
    openingHours: {
      start: {
        type: String,
        default: '00:00',
      },
      end: {
        type: String,
        default: '23:59',
      },
    },
    beds: {
      total: {
        type: Number,
        default: 0,
      },
      available: {
        type: Number,
        default: 0,
      },
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

// Indexes
HospitalSchema.index({ 'address.city': 1, emergencyServices: 1 });
HospitalSchema.index({ specialties: 1 });

export const Hospital = mongoose.model<IHospital>('Hospital', HospitalSchema);
